### DISCUSSION TITLE
Logistic Regression

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#544] [Q&A] [answered]
Dear team Algoritma,

Saya mendapatkan permasalahan untuk pemodelan menggunakan logistic regression. Dilihat dari proporsi berdasarkan variabel target, hasilnya tidak balance (99% = 'N' dan 1% = 'Y'). 

<img width="202" alt="image" src="https://user-images.githubusercontent.com/103427989/219285005-35e81653-03e7-46d8-8f54-9f407e934131.png">

Opsi apa yang sebaiknya dilakukan ya, 
1. apakah melakukan downsampling pada data mayoritas atau upsampling pada minoritas.
2. Melakukan klasifikasi terlebih dahulu sebelum log.regression

Mohon sarannya

Best regards,
JT

**Comment by finesaaa** [comment]
Dear @joejnt 

Terima kasih atas pertanyaan yang disampaikan, pertanyaan Anda sudah kami terima. Kami akan mencoba membantu mencarikan referensi berkaitan dengan pertanyaan ini.

**Comment by ranydwi** [comment, marked as answer]
Dear @joejnt 

Terima kasih untuk pertanyaan yang masuk, untuk menangani kasus yang bapak hadapai yaitu extremely imbalance classification, beberapa yang bapak bisa lakukan diantaranya:
1. Melakukan **teknik undersampling/oversampling tradisional** dimana hal ini tentunya memiliki sedikit kekurangan ketika kita menghadapi data yang sangat tidak balance. 
- Upsampling: metode ini memperbesar ukuran kelas minoritas dengan pengambilan sampel dengan penggantian sehingga kelas-kelas tersebut akan memiliki ukuran yang sama 
- Downsampling: mengurangi ukuran kelas mayoritas menjadi sama atau mendekati ukuran kelas minoritas hanya dengan mengambil sampel secara acak.

2.  Melakukan **hybrid method** yang lebih disarankan dikarenakan metode ini akan menghasilkan sampel yang _synthetic balanced_. Beberapa metode yang bisa dilakukan di R:
- `ROSE` atau Random Over-Sampling Examples
- `SMOTE` atau Synthetic Minority Oversampling Technique

Referensi yang bisa Bapak baca:
- Masalah berikut sangat serupa dengan yang bapak alami yaitu extreme imbalance dataset: https://www.analyticsvidhya.com/blog/2016/03/practical-guide-deal-imbalanced-classification-problems/
- Dokumentasi `ROSE`: https://rdrr.io/cran/ROSE/man/ROSE-package.html
- Dokumentasi `SMOTE`: https://www.statology.org/smote-in-r/

Semoga membantu :)
Regards.

**Reply by joejnt** [reply]
dear @ranydwi 

Terima kasih untuk inside nya.

Best regards,
JT
